src/App.jsx

```diff
@@ -14,45 +16,96 @@ export const goodsFromServer = [
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const LENGTH = 'length';
+const ALPHABET = 'alphabet';
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case ALPHABET:
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case LENGTH:
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        return 0;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortField, setSortField] = useState('');
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== ALPHABET,
+          })}
+          onClick={(event) => {
+            setSortField(ALPHABET);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== LENGTH,
+          })}
+          onClick={() => {
+            setSortField(LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': isReversed === false,
+          })}
+          onClick={() => {
+            setReversed(!isReversed);
+          }}
+        >
+          Reverse
+        </button>
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className={classNames('button', 'is-danger', 'is-light')}
+            onClick={() => {
+              setSortField('');
+              setReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+      {visibleGoods.map(good => (
+        <ul key={good}>
+          <li data-cy="Good">{good}</li>
+        </ul>
+      ))}
```

Comment:
looks wrong
<img width="446" alt="image" src="https://github.com/mate-academy/react_list-of-goods-js/assets/70220667/624c6a89-d95d-4c11-9668-45cdf23a25ea">
```suggestion
     <ul >
        {visibleGoods.map(good => (
          <li key={good} data-cy="Good">{good}</li>
         ))}
      </ul>
```